americangut/util.py

```diff
@@ -635,3 +636,58 @@ def get_single_id_lists(map_, depths):
             single_ids[depths[idx - 1]].extend(single_ids[depths[idx]])
 
     return single_ids
+
+
+def collapse_taxonomy(_bt, level=5):
+    """Collapses OTUs by taxonomy
+
+    Parameters
+    ----------
+    _bt : biom table
+        Table to collapse
+    level : int, optional
+        Level to collapse to. 0=kingdom, 1=phylum,...,5=genus, 6=species
+        Default 5
+
+    Returns
+    -------
+    biom table
+        Collapsed biom table
+
+    Citations
+    ---------
+    [1] http://biom-format.org/documentation/table_objects.html
+    """
+    def collapse_f(id_, md):
+        return '; '.join(md['taxonomy'][:level + 1])
+    collapsed = _bt.collapse(collapse_f, axis='observation', norm=False)
+    return collapsed
+
+
+def collapse_full(_bt):
```

Comment:
You might be able to do it with `collapse` but expressing `median` in that context will be weird as it would need to be a rolling median -- iirc, you can control the function that describes how values are smashed together, or it's possible we took that out as there was performance overhead. But, I think there is a more direct route:

```python
return Table([np.median(v) for v in _bt.iter_data(axis='observation')], 
             _bt.ids(axis='observation'),
             observation_metadata=_bt.metadata(axis='observation'))
```

Unfortunately, because of the rolling issue, you can't do the "cleanest" thing which would be a `reduce`:

```python
# won't work... :(
return Table.reduce(np.median, axis='observation')
```